Sources/TunnelKitWireGuardAppExtension/WireGuardTunnelProvider.swift

```diff
@@ -146,6 +175,38 @@ extension WireGuardTunnelProvider {
         // store path for clients
         cfg._appexSetDebugLogPath()
     }
+
+    func getStats() throws -> WireGuardDataCount {
+        var result: String?
+
+        let dispatchGroup = DispatchGroup()
+        dispatchGroup.enter()
+        adapter.getRuntimeConfiguration { string in
+            result = string
+            dispatchGroup.leave()
+        }
+
+        guard case .success = dispatchGroup.wait(wallTimeout: .now() + 1) else { throw StatsError.timeout }
+        guard let result else { throw StatsError.nilValue }
+        guard let newStats = WireGuardDataCount(from: result) else { throw StatsError.parse }
```

Comment:
Blocking code is a no-go. Please drop any of this DispatchGroup stuff and make this method callback-based instead. In the end, refreshDataCount() does not need the result to be synchronous.

```swift
func fetchDataCount(completiondHandler: (Result<WireGuardDataCount, Error>)) {
    ...
}
```

Reply:
done